src/main/java/org/jabref/logic/cleanup/RenamePdfCleanup.java

```diff
@@ -150,28 +137,24 @@ public RenamePdfCleanup(boolean onlyRelativePaths, BibDatabaseContext databaseCo
             //we put an undo of the field content here
             //the file is not being renamed back, which leads to inconsistencies
             //if we put a null undo object here, the change by "doMakePathsRelative" would overwrite the field value nevertheless.
-            if (change.isPresent()) {
-                return Collections.singletonList(change.get());
-            } else {
-                return Collections.emptyList();
-            }
+            return change.map(Collections::singletonList).orElseGet(Collections::emptyList);
         }
         return Collections.emptyList();
     }
 
     public String getTargetFileName(LinkedFile flEntry, BibEntry entry) {
         String realOldFilename = flEntry.getLink();
 
-        StringBuilder targetFileName = new StringBuilder(FileUtil
-                .createFileNameFromPattern(databaseContext.getDatabase(), entry, fileNamePattern, layoutPrefs)
-                .trim());
-        //Add extension to newFilename
-        targetFileName.append('.').append(FileHelper.getFileExtension(realOldFilename).orElse("pdf"));
-        return targetFileName.toString();
+        String targetFileName = FileUtil.createFileNameFromPattern(
```

Comment:
Why do you drop the StringBuilder here?

Reply:
Just normal String concatenation. I thin, the StringBuilder did not introduce more readability. In the current code, I find it more readable. But, maybe, I am a 2005 programmer 🎉

Reply:
As I just read the String concat gets nonetheless optimized as a StringBuilder on the compiler side, so I guess it's safe here. 

At least when doing concats in loops you should stick to StringBuilder because its more efficient

Reply:
I don't see the point of using a StringBuilder there either, it feels like it clutters the code. Used String.format() as an alternative before, but I think this is the most readable variant.